lib/segment/src/vector_storage/multi_dense/appendable_mmap_multi_dense_vector_storage.rs

```diff
@@ -181,6 +181,15 @@ impl<T: PrimitiveVectorElement> VectorStorage for AppendableMmapMultiDenseVector
         self.offsets.len()
     }
 
+    fn available_size_in_bytes(&self) -> usize {
+        let total_size = self.vectors.len() * self.vector_dim() * std::mem::size_of::<T>();
```

Comment:
Another `saturating_mul` candidate?

Reply:
here we cannot overflow usize because it's a size in bytes of storage. But we can overflow it next line, I added `u128` usage. The reason is the same as here https://github.com/qdrant/qdrant/pull/4312#discussion_r1616029582